### DISCUSSION TITLE
Resetting form after successful submission and post

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#9724] [Q&A]
After reading the docs I realised I've been resetting my form incorrectly, although it seems to work without any problems:
```ts
  const onSubmitHandler: SubmitHandler<MyInputs> = async (
    data
  ) => {
    const result = await postData(data);

    if ("error" in result) {
      showErrorNotification({ message: t("The changes couldn't be saved") });
    } else {
      form.reset({ name: null });
      showSuccessNotification({ message: t("The changes are saved") });
    }
  };
```

But in the sandbox linked from within the docs it resets the form after a successful submission although it can be successful in the client but if the api request for some reason fails I don't want to reset the form. Do you see the problem? 

https://codesandbox.io/s/react-hook-form-handlesubmit-with-reset-xu1zu

**Comment by bluebill1049** [comment]
Could you set that at useEffect with `isSubmitSuccessful` as dep?
